korangar/src/loaders/model/mod.rs

```diff
@@ -257,23 +257,43 @@ impl ModelLoader {
         )
     }
 
-    pub fn calculate_transformation_matrix(node: &mut Node, is_root: bool, bounding_box: AABB, parent_matrix: Matrix4<f32>) {
-        node.transform_matrix = match is_root {
+    pub fn calculate_transformation_matrix(
+        node: &mut Node,
+        is_root: bool,
+        bounding_box: AABB,
+        parent_matrix: Matrix4<f32>,
+        is_static: bool,
+    ) {
+        let transform_matrix = match is_root {
             true => {
                 let translation_matrix = Matrix4::from_translation(-Vector3::new(
                     bounding_box.center().x,
                     bounding_box.max().y,
                     bounding_box.center().z,
                 ));
 
-                parent_matrix * translation_matrix * node.transform_matrix
+                translation_matrix * node.transform_matrix
             }
-            false => parent_matrix * node.transform_matrix,
+            false => node.transform_matrix,
+        };
+        node.transform_matrix = match is_static {
+            true => parent_matrix * transform_matrix,
+            false => transform_matrix,
         };
 
-        node.child_nodes
-            .iter_mut()
-            .for_each(|child_node| Self::calculate_transformation_matrix(child_node, false, bounding_box, node.transform_matrix));
+        node.child_nodes.iter_mut().for_each(|child_node| {
+            Self::calculate_transformation_matrix(child_node, false, bounding_box, node.transform_matrix, is_static);
+        });
+    }
+
+    // A model is static, if it doesn't have any animations.
+    pub fn is_static(node: &Node) -> bool {
+        let mut is_static = node.rotation_keyframes.is_empty();
+        node.child_nodes.iter().for_each(|child_node| {
+            let is_static_child = Self::is_static(child_node);
+            is_static &= is_static_child;
+        });
+        is_static
```

Comment:
This can be expressed like

```rust
        node.rotation_keyframes.is_empty() && node.child_nodes.iter().all(Self::is_static)
```

The advantages here are
- no `mut` required
- the function can exit early now as soon as something dynamic is found